src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -256,8 +256,7 @@ private static boolean shouldBeAloneOnLineWithAloneOption(Details details) {
      *         when ALONE_OR_SINGLELINE option is used.
      */
     private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details) {
-        return !isAloneOnLine(details)
```

Comment:
This became a new mutation when the other code was removed inside `isSingleLineBlock`. I am unsure on the exact reasoning but I believe it is because other similar code handles the same thing. I wasn't able to produce anything to keep it and regression showed nothing.
With this gone, I felt it best to rename `isSingleLineBlock` to `isRightCurlyOnSameLineAsNextToken`.